### PR TITLE
Update e2e test to run against machine.openshift.io

### PR DESCRIPTION
**Pull request by enxebre**
Update e2e test to run against machine.openshift.io
Needs https://github.com/openshift/installer/pull/1175

### CODE REVIEW

File: test/e2e/main.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func runSuite() error {
 	glog.Info("PASS: ExpectProviderAvailable")
 
 	glog.Info("RUN: ExpectOneClusterObject")
```
Comment (ingvagabund) [inline test/e2e/main.go]: `s/ExpectOneClusterObject/ExpectNoClusterObject`

Comment (enxebre) [reply test/e2e/main.go]: we don't need the cluster for machine.openshift.io https://github.com/openshift/cluster-api-provider-aws/pull/151/files#diff-1c49fed4c53a7b1e61a888973c253576R54

Comment (frobware) [reply test/e2e/main.go]: Longer term we should have something more sophisticated so that we don't have to repeat all these string values.

**Pull request merged**